app/src/features/surveys/view/survey-spatial/components/animal/SurveySpatialAnimal.tsx

```diff
@@ -25,28 +22,23 @@ interface ISurveySpatialAnimalProps {
  * @param {ISurveySpatialAnimalProps} props
  * @returns
  */
-export const SurveySpatialAnimal = (props: ISurveySpatialAnimalProps) => {
+export const SurveySpatialAnimal = (props: ISurveySpatialAnimalMapProps) => {
   const surveyContext = useSurveyContext();
-
   const crittersApi = useCritterbaseApi();
 
   const critterIds = useMemo(
     () => surveyContext.critterDataLoader.data?.map((critter) => critter.critterbase_critter_id) ?? [],
     [surveyContext.critterDataLoader.data]
   );
 
-  // Data loader for fetching animal capture data for the map ONLY. Table data is fetched separately in `SurveySpatialAnimalTable.tsx`
   const geometryDataLoader = useDataLoader((critter_ids: string[]) =>
     crittersApi.critters.getMultipleCrittersGeometryByIds(critter_ids)
   );
 
   useEffect(() => {
-    if (!critterIds.length) {
-      return;
+    if (critterIds.length) {
+      geometryDataLoader.load(critterIds);
     }
-
-    geometryDataLoader.load(critterIds);
-    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [critterIds]);
```

Comment:
Missing dependency: `geometryDataLoader`

Reply:
Its possible that including it will result in an infinite loop. In which case we need to include that comment `// eslint-disable-next-line react-hooks/exhaustive-deps` which tells the linter to ignore the fact that there is a missing dependency.